src/app/organization/working-days/working-days.component.ts

```diff
@@ -0,0 +1,113 @@
+  templateUrl: './working-days.component.html',
+  styleUrls: ['./working-days.component.scss']
+})
+export class WorkingDaysComponent implements OnInit {
+
+  /** Working days data. */
+  workingDaysData: any;
+  /** Week days to be used for checkboxes */
+  weekDays = [
+    {name: 'MO', value: 'Monday', checked: false},
+    {name: 'TU', value: 'Tuesday', checked: false},
+    {name: 'WE', value: 'Wednesday', checked: false},
+    {name: 'TH', value: 'Thursday', checked: false},
+    {name: 'FR', value: 'Friday', checked: false},
+    {name: 'SA', value: 'Saturday', checked: false},
+    {name: 'SU', value: 'Sunday', checked: false}
+  ];
+  /** Working days form. */
+  workingDaysForm: FormGroup;
+  /** Data source for repayment reschedule type select */
+  repaymentRescheduleOptionsData: any;
+
+  /**
+   * Retrieves the working days data from `resolve`.
+   * @param {formBuilder} formBuilder form builder.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {organizationService} organizationService organization service.
+   * @param {Router} router Router for navigation
+   */
+  constructor(private formBuilder: FormBuilder,
+              private route: ActivatedRoute,
+              private organizationService: OrganizationService,
+              private router: Router) {
+    this.route.data.subscribe((data: { workingDays: any }) => {
+      this.workingDaysData = data.workingDays;
+    });
+  }
+
+  /**
+   * Creates the working days form.
+   */
+  ngOnInit() {
+    this.createWorkingDaysForm();
+  }
+
+  /**
+   * Creates the working days form.
+   */
+  createWorkingDaysForm() {
+    this.setWeekWorkingDays();
+    this.repaymentRescheduleOptionsData = this.workingDaysData.repaymentRescheduleOptions;
+    this.workingDaysForm = this.formBuilder.group({
+      'recurrence': this.formBuilder.array(this.createRecurrenceFormArray()),
+      'repaymentRescheduleType': [this.workingDaysData.repaymentRescheduleType.id],
+      'extendTermForDailyRepayments': [this.workingDaysData.extendTermForDailyRepayments]
+    });
+  }
+
+  /**
+   * Initialize days checkboxes
+   */
+  setWeekWorkingDays() {
+    const days = this.workingDaysData.recurrence.replace('FREQ=WEEKLY;INTERVAL=1;BYDAY=', '');
+    for (let i = 0; i < this.weekDays.length; i++) {
+      this.weekDays[i].checked = days.includes(this.weekDays[i].name);
+    }
+  }
+
+  /**
+   * Creates the recurrence form array.
+   */
+  createRecurrenceFormArray() {
+    const recurrence: any = [];
+    for (let i = 0; i < this.weekDays.length; i++) {
+      const day = this.formBuilder.control(i);
+      day.setValue(this.weekDays[i].checked);
+      recurrence.push(day);
+    }
+    return recurrence;
+  }
+
+  /**
+   * Edits the current working days configuration,
+   * if successful redirects to organization view.
+   */
+  submit() {
+    const workingDays = this.workingDaysForm.value;
+    let recurrence = 'FREQ=WEEKLY;INTERVAL=1;BYDAY=';
```

Comment:
Create a constant above the class declaration and use it instead. (you can use it here as well as above in the replace function instead of writing the same string again)